gef.py

```diff
@@ -736,270 +714,364 @@ class Elf:
-    SHT_SYMTAB           = 2
-    SHT_STRTAB           = 3
-    SHT_RELA             = 4
-    SHT_HASH             = 5
-    SHT_DYNAMIC          = 6
-    SHT_NOTE             = 7
-    SHT_NOBITS           = 8
-    SHT_REL              = 9
-    SHT_SHLIB            = 10
-    SHT_DYNSYM           = 11
-    SHT_NUM	             = 12
-    SHT_INIT_ARRAY       = 14
-    SHT_FINI_ARRAY       = 15
-    SHT_PREINIT_ARRAY    = 16
-    SHT_GROUP            = 17
-    SHT_SYMTAB_SHNDX     = 18
-    SHT_LOOS             = 0x60000000
-    SHT_GNU_ATTRIBUTES   = 0x6ffffff5
-    SHT_GNU_HASH         = 0x6ffffff6
-    SHT_GNU_LIBLIST      = 0x6ffffff7
-    SHT_CHECKSUM         = 0x6ffffff8
-    SHT_LOSUNW           = 0x6ffffffa
-    SHT_SUNW_move        = 0x6ffffffa
-    SHT_SUNW_COMDAT      = 0x6ffffffb
-    SHT_SUNW_syminfo     = 0x6ffffffc
-    SHT_GNU_verdef       = 0x6ffffffd
-    SHT_GNU_verneed      = 0x6ffffffe
-    SHT_GNU_versym       = 0x6fffffff
-    SHT_HISUNW           = 0x6fffffff
-    SHT_HIOS             = 0x6fffffff
-    SHT_LOPROC           = 0x70000000
-    SHT_HIPROC           = 0x7fffffff
-    SHT_LOUSER           = 0x80000000
-    SHT_HIUSER           = 0x8fffffff
-
-    SHF_WRITE            = 1
-    SHF_ALLOC            = 2
-    SHF_EXECINSTR        = 4
-    SHF_MERGE            = 0x10
-    SHF_STRINGS          = 0x20
-    SHF_INFO_LINK        = 0x40
-    SHF_LINK_ORDER       = 0x80
-    SHF_OS_NONCONFORMING = 0x100
-    SHF_GROUP            = 0x200
-    SHF_TLS              = 0x400
-    SHF_COMPRESSED       = 0x800
-    SHF_RELA_LIVEPATCH   = 0x00100000
-    SHF_RO_AFTER_INIT    = 0x00200000
-    SHF_ORDERED          = 0x40000000
-    SHF_EXCLUDE          = 0x80000000
-
-    sh_name: Union[int, str] = 0
-    sh_type: int             = 0
-    sh_flags: int            = 0
-    sh_addr: int             = 0
-    sh_offset: int           = 0
-    sh_size: int             = 0
-    sh_link: int             = 0
-    sh_info: int             = 0
-    sh_addralign: int        = 0
-    sh_entsize: int          = 0
+    class Type(enum.IntEnum):
+        SHT_NULL             = 0
+        SHT_PROGBITS         = 1
+        SHT_SYMTAB           = 2
+        SHT_STRTAB           = 3
+        SHT_RELA             = 4
+        SHT_HASH             = 5
+        SHT_DYNAMIC          = 6
+        SHT_NOTE             = 7
+        SHT_NOBITS           = 8
+        SHT_REL              = 9
+        SHT_SHLIB            = 10
+        SHT_DYNSYM           = 11
+        SHT_NUM	             = 12
+        SHT_INIT_ARRAY       = 14
+        SHT_FINI_ARRAY       = 15
+        SHT_PREINIT_ARRAY    = 16
+        SHT_GROUP            = 17
+        SHT_SYMTAB_SHNDX     = 18
+        SHT_LOOS             = 0x60000000
+        SHT_GNU_ATTRIBUTES   = 0x6ffffff5
+        SHT_GNU_HASH         = 0x6ffffff6
+        SHT_GNU_LIBLIST      = 0x6ffffff7
+        SHT_CHECKSUM         = 0x6ffffff8
+        SHT_LOSUNW           = SHT_SUNW_move       = 0x6ffffffa
```

Comment:
this syntax is shorter, but maybe also less concise? I don't have a strong opinion on this but I noticed it while reading and wanted to put it out there before this is merged. (same comment applies for the two other places like this further down...)

Reply:
I am not a fan of this style either.
I'd rather something like
```suggestion
        SHT_LOSUNW           = 0x6ffffffa
        SHT_SUNW_move        = SHT_LOSUNW
```